components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -929,7 +929,7 @@ private static String getScopeMetadataQueryParam(Set<String> scopes, String tena
             String scopeMetadata = new Gson().toJson(scopesMetaData);
             return "scopeMetadata=" + URLEncoder.encode(scopeMetadata, UTF_8);
         } catch (Exception e) {
-            log.warn("Error while retrieving scope metadata for scopes: " + scopes, e);
+            log.debug("Error while retrieving scope metadata for scopes: " + scopes, e);
```

Comment:
since we are doing string concatenation better to check isDebugEnabled before printing the debug log.

Reply:
addressed with [https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2051/commits/e389849b36b23d92e7c4a15a6d477a42218bc21f](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2051/commits/e389849b36b23d92e7c4a15a6d477a42218bc21f)